fat/fat2/transactionbatch.go

```diff
@@ -0,0 +1,141 @@
+package fat2
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/Factom-Asset-Tokens/factom"
+	"github.com/Factom-Asset-Tokens/fatd/fat"
+)
+
+// Transaction represents a fat2 transaction, which can be a list of one or more
+// sub-transactions to be executed in order
+type TransactionBatch struct {
+	Version      uint          `json:"version"`
+	Transactions []Transaction `json:"transactions"`
+	fat.Entry
+}
+
+// NewTransactionBatch returns a TransactionBatch initialized with the given entry.
+func NewTransactionBatch(entry factom.Entry) TransactionBatch {
+	return TransactionBatch{Entry: fat.Entry{Entry: entry}}
+}
+
+type transactionBatch TransactionBatch
+
+// MarshalJSON marshals the TransactionBatch content field as JSON, but will raise an error if
+// the batch fails the checks in ValidData()
+func (t TransactionBatch) MarshalJSON() ([]byte, error) {
+	if err := t.ValidData(); err != nil {
+		return nil, err
+	}
+	return json.Marshal(transactionBatch(t))
+}
+
+func (t TransactionBatch) String() string {
+	data, err := t.MarshalJSON()
+	if err != nil {
+		return err.Error()
+	}
+	return string(data)
+}
+
+// UnmarshalEntry unmarshals the Entry content as a TransactionBatch
+func (t *TransactionBatch) UnmarshalEntry() error {
+	return t.Entry.UnmarshalEntry(t)
+}
+
+// MarshalEntry marshals the TransactionBatch into the entry content
+func (t *TransactionBatch) MarshalEntry() error {
+	return t.Entry.MarshalEntry(t)
+}
+
+// Valid performs all validation checks and returns nil if it is a valid batch. This function assumes
+// the struct's entry field is populated.
+func (t *TransactionBatch) Valid() error {
```

Comment:
Please rename this to `Validate()` to conform to the new naming scheme.

Validate is a verb and returns an error which makes more sense than the noun Valid.

Reply:
All instances of `Valid*()` to `Validate*()` correct?

Reply:
Nevermind. Looks like only the Valid() function. Committing now